common/components/componentsBySection/FindProjects/Filters/RenderFilterCategory.jsx

```diff
@@ -0,0 +1,127 @@
+
+    //group by subcategories, then sort and map just like parent component but for subcategories
+    let groupedSubcats = _.groupBy(this.props.data, 'subcategory');
+    let sortedKeys = Object.keys(groupedSubcats).sort(); //default sort is alphabetical, what we want
+    let categoryKey = this.props.category; //for the whole category's container element
+    const displaySubcategories = sortedKeys.map(key =>
+          <div className={this.state[key] ? classSubcategoryExpanded : classSubcategoryCollapsed} key={key}>
+            <div className="ProjectFilterContainer-subcategory-header"  id={key} onClick={(e) => this._handleChange(key, e)}>
+              <span>{key}</span><span className="ProjectFilterContainer-showtext">{this.state[key] ? <i className={GlyphStyles.ChevronUp}></i> : <i className={GlyphStyles.ChevronDown}></i>}</span>
+            </div>
+            <div className="ProjectFilterContainer-content">
+              {this._renderFilterList(groupedSubcats[key])}
+            </div>
+          </div>
+        );
+      return (
+        <div className={this.state[categoryKey] ? classCategoryExpanded : classCategoryCollapsed} key={categoryKey}>
+          <div className="ProjectFilterContainer-category-header" id={this.props.category} onClick={(e) => this._handleChange(categoryKey, e)}>
+            <span>{this._displayName(this.props.category)}</span>
+            <span className="ProjectFilterContainer-showtext">{this.state[categoryKey] ? <i className={GlyphStyles.ChevronUp}></i> : <i className={GlyphStyles.ChevronDown}></i>}</span>
+          </div>
+          <div className="ProjectFilterContainer-content">
+            {displaySubcategories}
+          </div>
+        </div>
+      )
+  }
+  _renderNoSubcategories() {
+    let categoryKey = this.props.category;
+    //if a category has NO subcategories (hasSubcategories is false), render a single list
+    return (
+        <div className={this.state[categoryKey] ? classCategoryExpanded : classCategoryCollapsed}>
+          <div className='ProjectFilterContainer-category-header' id={categoryKey} onClick={(e) => this._handleChange(categoryKey, e)}>
+            <span>{this._displayName(this.props.category)}</span>
+            <span className="ProjectFilterContainer-showtext">{this.state[categoryKey] ? <i className={GlyphStyles.ChevronUp}></i> : <i className={GlyphStyles.ChevronDown}></i>}</span>
+          </div>
+          <div className="ProjectFilterContainer-content">
+            {this._renderFilterList(this.props.data)}
+          </div>
+        </div>
+    );
+  }
+
+  _renderFilterList(data) {
+    //this function renders individual clickable filter items regardless of category or subcategory status
+    let sortedTags = Object.values(data).map((tag) =>
+      <li key={tag.category + '-' + tag.display_name} className="ProjectFilterContainer-list-item">
```

Comment:
The expression
tag.category + '-' + tag.display_name
is repeated multiple times.  It should be a const.